### PR TITLE
Extend spectrum_from_signal and signal_from_spectrum to multichannel signals

### PR DESCRIPTION
**Pull request by fietew**
Try:
```MATLAB
SFS_start;
conf = SFS_config;
conf.plot.useplot = true;
SOFAstart;

irs = get_ir(dummy_irs(1024,conf),[0,0,0],0,[1 0 0], 'cartesian',conf);
[b1,a1] = butter(5,0.1);
[b2,a2] = butter(10,0.2);

sig1 = filter(b1,a1,irs(:,1));
sig2 = filter(b2,a2,irs(:,1));
sig = [irs(:,1), sig1, sig2];

[amp, phase, f] = spectrum_from_signal(sig,1,conf);
sig_inv = signal_from_spectrum(amp, phase, f, 1,conf);
figure, 
plot(sig,'k');
hold on
plot(sig_inv, '--');
hold off;

[amp, phase, f] = spectrum_from_signal(sig,conf);
sig_inv = signal_from_spectrum(amp, phase, f, conf);
figure, 
plot(sig,'k');
hold on
plot(sig_inv, '--');
hold off;

[amp, phase, f] = spectrum_from_signal(sig.',2,conf);
sig_inv = signal_from_spectrum(amp, phase,f,2,conf);
figure, 
plot(sig,'k');
hold on
plot(sig_inv.', '--');
hold off;
```

### CODE REVIEW

File: SFS_general/signal_from_spectrum.m [hunk 1]
```diff
@@ -62,30 +77,40 @@
...
-    phase = [phase; -1*phase(end:-1:2)];
+    phase = [phase; -1*phase(end:-1:2,:)];
 end
 
 % Convert to complex spectrum
 compspec = amplitude .* exp(1i*phase);
 
 % Build the inverse fft and assume spectrum is conjugate symmetric
 outsig = real(ifft(compspec));
+
```
Comment (hagenw) [inline SFS_general/signal_from_spectrum.m]: Add another empty line here.

File: SFS_general/spectrum_from_signal.m [hunk 2]
```diff
@@ -1,10 +1,11 @@
-function varargout = spectrum_from_signal(signal,conf)
+function varargout = spectrum_from_signal(signal,dim,conf)
 %SPECTRUM_FROM_SIGNAL single-sided amplitude and phase spectra of signal
 %
-%   Usage: [amplitude,phase,f] = spectrum_from_signal(sig,conf)
+%   Usage: [amplitude,phase,f] = spectrum_from_signal(sig,[dim],conf)
 %
 %   Input parameters:
 %       signal      - one channel audio (time) signal
```
Comment (hagenw) [inline SFS_general/spectrum_from_signal.m]: Update description as we can have now multiple channels.

File: SFS_general/spectrum_from_signal.m [hunk 3]
```diff
@@ -65,47 +77,57 @@
...
     figure; title('Spectrum');
     subplot(2,1,1)
     semilogx(f,20 * log10(abs(amplitude))); xlim([1, fs/2]);
     grid on; xlabel('frequency / Hz'); ylabel('amplitude / dB')
     subplot(2,1,2)
-    semilogx(f,unwrap(phase)); xlim([1, fs/2]);
+    semilogx(f,unwrap(phase,[],1)); xlim([1, fs/2]);
     grid on; xlabel('frequency / Hz'); ylabel('phase / rad')
 end
+
```
Comment (hagenw) [inline SFS_general/spectrum_from_signal.m]: Add another empty line here.

**Comment by chris-hld** [comment]
Looks good!
Should we adjust the test script?

**Comment by hagenw** [comment]
Should be fine without adjusting the test script, I will merge this now.

**Pull request merged**